utils/src/networking.rs

```diff
@@ -333,4 +371,123 @@ mod tests {
         let addr_v6 = NetAddress::from_str("[2a01:4f8:191:1143::2]:5678");
         assert!(addr_v6.is_ok());
     }
+
+    #[test]
+    fn test_is_publicly_routable() {
+        // RFC 2544 tests
+        assert_eq!(false, IpAddress::from_str("198.18.0.0").unwrap().is_publicly_routable());
```

Comment:
clippy warns that assert should be used instead of assert_eq with a literal 

for all `assert_eq!(false, ..` -> `assert!(!..`
for all `assert_eq!(true, ..` -> `assert!(..`